vault/identity_store_oidc_provider.go

```diff
@@ -1384,6 +1386,17 @@ func (i *IdentityStore) pathOIDCAuthorize(ctx context.Context, req *logical.Requ
 		return authResponse("", state, ErrAuthInvalidRequest, "nonce parameter is required")
 	}
 
+	// We don't support the request or request_uri parameters. If they're provided,
+	// the appropriate errors must be returned. For details, see the spec at:
+	// https://openid.net/specs/openid-connect-core-1_0.html#RequestObject
+	// https://openid.net/specs/openid-connect-core-1_0.html#RequestUriParameter
+	if _, ok := d.Raw["request"]; ok {
+		return authResponse("", state, ErrAuthRequestNotSupported, "request parameter is not supported")
+	}
+	if _, ok := d.Raw["request_uri"]; ok {
```

Comment:
I was looking into the spec to understand this better and I think we may need to include the `request_uri_parameter_supported=false` in the Discovery endpoint response. The spec states that the default is `true` if it is not included. https://openid.net/specs/openid-connect-discovery-1_0.html#ProviderMetadata

Reply:
Nice catch! Added in the other PR https://github.com/hashicorp/vault/pull/12623.